### PR TITLE
Improve fetch api code

### PR DESCRIPTION
**Pull request by magicznyleszek**
## Checklist

1. [ ] If you've added code that should be tested, add tests
2. [ ] If you've changed APIs, update (or create!) the documentation
3. [x] Ensure the tests pass
4. [x] Make sure that your code lints and that you've followed [our coding style](https://github.com/kobotoolbox/kpi/blob/master/CONTRIBUTING.md)
5. [x] Write a title and, if necessary, a description of your work suitable for publishing in our [release notes](https://community.kobotoolbox.org/tag/release-notes)
6. [ ] Mention any related issues in this repository (as #ISSUE) and in other repositories (as kobotoolbox/other#ISSUE)
7. [ ] Open an issue in the [docs](https://github.com/kobotoolbox/docs/issues/new) if there are UI/UX changes

## Description

Removed duplicated error notification and some internal code improvements.

## Notes

This is needed for #4645 and in general to make it possible to avoid duplicated error notifications (impossible without this change).

### CODE REVIEW

File: jsapp/js/api.ts [hunk 1]
```diff
@@ -98,24 +123,47 @@ const fetchData = async <T>(
...
-  fetchData<T>(url, 'POST', data, false);
+export const fetchPostUrl = async <T>(
+  url: string,
+  data: Json,
+  options?: FetchDataOptions
+) => {
+  if (!options) {
+    options = {};
+  }
+  options.prependRootUrl = false;
```
Comment (p2edwards) [inline jsapp/js/api.ts:145]:
```ts
options = Object.assign({}, options, {prependRootUrl: false})
```

**Review by p2edwards — COMMENTED** [review]
Reviewed the code and made some preliminary suggestions.

I want to manually test this before approving.

File: jsapp/js/api.ts [hunk 2]
```diff
@@ -16,21 +16,49 @@ const JSON_HEADER = 'application/json';
...
 ) => {
+  // Prepare options
+  let notifyAboutError = true;
+  if (typeof options?.notifyAboutError === 'boolean') {
+    notifyAboutError = options.notifyAboutError;
+  }
+  let prependRootUrl = true;
+  if (typeof options?.prependRootUrl === 'boolean') {
+    prependRootUrl = options.prependRootUrl;
+  }
```
[the remark this answers is not in the document]
Comment (magicznyleszek) [reply jsapp/js/api.ts]: I like it :)

**Comment by p2edwards** [comment]
Merging now as it's an improvement. #4645 will warrant comprehensive testing for the next release.

**Pull request merged**